tests/src/analysis/testqgsprocessingsfcgalalgs.cpp

```diff
@@ -0,0 +1,157 @@
+  mPolygonLayer = new QgsVectorLayer( polyFileInfo.filePath(), QStringLiteral( "polygons" ), QStringLiteral( "ogr" ) );
+  QVERIFY( mPolygonLayer->isValid() );
+
+  // Register the layer with the registry
+  QgsProject::instance()->addMapLayer( mPolygonLayer );
+}
+
+void TestQgsProcessingSfcgalAlgs::cleanupTestCase()
+{
+  QgsProject::instance()->removeMapLayer( mPolygonLayer );
+  QgsApplication::exitQgis();
+}
+
+void TestQgsProcessingSfcgalAlgs::init()
+{
+}
+
+void TestQgsProcessingSfcgalAlgs::cleanup()
+{
+}
+
+QgsGeometry TestQgsProcessingSfcgalAlgs::openWktFile( const QString &wktFile )
+{
+  QString expectedPath = testDataPath( QString( "control_files/expected_sfcgal/expected_%1" ).arg( wktFile ) );
+  QFile expectedFile( expectedPath );
+  if ( !expectedFile.open( QFile::ReadOnly | QIODevice::Text ) )
+  {
+    qWarning() << "Unable to open expected data file" << expectedPath;
+    return QgsGeometry();
+  }
+
+  // remove '\n' from dumped file
+  QByteArray expectedBA = expectedFile.readAll();
+  QString expectedStr;
+  for ( int i = 0; i < expectedBA.length(); i++ )
+  {
+    if ( expectedBA.at( i ) != '\n' )
+      expectedStr += expectedBA.at( i );
+  }
+
+  // load geom from corrected wkt
+  return QgsGeometry::fromWkt( expectedStr );
+}
+
+void TestQgsProcessingSfcgalAlgs::medialAxis()
+{
+  std::unique_ptr<QgsProcessingAlgorithm> alg( QgsApplication::processingRegistry()->createAlgorithmById( QStringLiteral( "native:approximatemedialaxis" ) ) );
+  QVERIFY( alg != nullptr );
+
+  auto context = std::make_unique<QgsProcessingContext>();
+  context->setProject( QgsProject::instance() );
+
+  QVariantMap parameters;
+  parameters.insert( QStringLiteral( "INPUT" ), QVariant::fromValue( mPolygonLayer ) );
+  parameters.insert( QStringLiteral( "OUTPUT" ), QgsProcessing::TEMPORARY_OUTPUT );
+
+  bool ok = false;
+  QgsProcessingFeedback feedback;
+  QVariantMap results = alg->run( parameters, *context, &feedback, &ok );
+  QVERIFY( ok );
+
+  QgsVectorLayer *outputLayer = qobject_cast<QgsVectorLayer *>( context->getMapLayer( results.value( QStringLiteral( "OUTPUT" ) ).toString() ) );
+  QVERIFY( outputLayer );
+  QVERIFY( outputLayer->isValid() );
+  QCOMPARE( outputLayer->featureCount(), 10 );
+
+  // retrieve first feature
+  QgsFeatureIterator outputFeatureIterator = outputLayer->dataProvider()->getFeatures();
+  QgsFeature outputFeature;
+  QVERIFY( outputFeatureIterator.nextFeature( outputFeature ) );
+
+  // check attribute
+  QGSCOMPARENEAR( outputFeature.attribute( "length" ).toDouble(), 14.01481, 0.00001 );
+
+  // check geometry
+  const QgsGeometry outputGeom( outputFeature.geometry() );
+  QVERIFY( outputGeom.wkbType() == Qgis::WkbType::MultiLineString );
+  const QgsGeometry expectedGeom = openWktFile( "processing_medial_axis.wkt" );
+  QVERIFY( expectedGeom.wkbType() == Qgis::WkbType::MultiLineString );
+
+  QVERIFY( outputGeom.constGet()->fuzzyEqual( *expectedGeom.constGet() ) );
```

Comment:
Let's compare via wkt instead, as that gives more useful debugging output if the tests fail then a simple boolean true/false result:

```suggestion
  QCOMPARE( outputGeom.constGet()->asWkt(4), expectedGeom.asWkt(4) ) );
```

(adjusting the precision as required)

Reply:
Done.